gammapy/maps/core.py

```diff
@@ -1088,6 +1116,63 @@ def reproject_to_geom(self, geom, preserve_counts=False, precision_factor=10):
             output_map = output_map.resample(geom, preserve_counts=preserve_counts)
         return output_map
 
+    def reproject_by_slice(
+        self,
+        geom,
+        preserve_counts=False,
+        precision_factor=10,
+    ):
+        """Reproject each slice of a 3d map to input 2d geometry.
+
+        Parameters
+        ----------
+        geom : `~gammapy.maps.Geom`
+            Target slice geometry (2d)
+        preserve_counts : bool
+            Preserve the integral over each bin.  This should be true
+            if the map is an integral quantity (e.g. counts) and false if
+            the map is a differential quantity (e.g. intensity)
+        precision_factor : int
+           Minimal factor between the bin size of the output map and the oversampled base map.
```

Comment:
This is not super clear for me....
You have a base map, then it is oversampled (there is a first bin factor). And then you add a second bin factor? 
I sure that I have not understand.

Reply:
The oversampling factor is not given by the user it is computed by dividing the `precision_factor` by the scale ratio of the output and input maps. Because the precision depends only of the factor between the oversampled map and the output map.